### PR TITLE
ExportRoute CRD  operations added

### PR DESCRIPTION
**Pull request by roopesh83**

### CODE REVIEW

File: cdl/export_route_service.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	Destination     Destination          `json:"destination" validate:"required"`
+	ServiceAccount  ExportServiceAccount `json:"serviceAccount" validate:"required"`
+	CreatedBy       string               `json:"createdBy,omitempty"`
+	CreatedOn       string               `json:"createdOn,omitempty"`
+	UpdatedBy       string               `json:"updatedBy,omitempty"`
+	UpdatedOn       string               `json:"updatedOn,omitempty"`
+}
+
+type Source struct {
+	CdlResearchStudy ExportResearchStudySource `json:"cdlResearchStudy" validate:"required"`
```
Comment (loafoe) [inline cdl/export_route_service.go:27]: Please rename to `CDLResearchStudy` as per `Go` convention

File: cdl/export_route_service.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+}
+
+type Destination struct {
+	CdlResearchStudy ExportResearchStudyDestination `json:"cdlResearchStudy" validate:"required"`
```
Comment (loafoe) [inline cdl/export_route_service.go:31]: `CDLResearchStudy`

File: cdl/export_route_service.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+	ExportLabel []ExportLabel `json:"label,omitempty"`
+}
+
+type ExportLabel struct {
+	Name             string `json:"name,omitempty"`
+	ApprovalRequired bool   `json:"approvalRequired,omitempty"`
+}
+
+type ExportServiceAccount struct {
+	CdlServiceAccount ExportServiceAccountDetails `json:"cdlServiceAccount" validate:"required"`
```
Comment (loafoe) [inline cdl/export_route_service.go:58]: `CDLServiceAccount`

File: cdl/export_route_service.go [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+}
+
+type ExportRouteBundleEntry struct {
+	FullURL  string      `json:"fullUrl,omitempty"`
+	Resource ExportRoute `json:"resource,omitempty"`
+}
+
+type LinkElementType struct {
+	Relation string `json:"relation"`
+	Url      string `json:"url"`
```
Comment (loafoe) [inline cdl/export_route_service.go:85]: `URL`

File: cdl/export_route_service.go [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+				lastPage = false
+			}
+		}
+		if lastPage {
+			return nil, resp, err
+		}
+	}
+}
+
+func (exp *ExportRouteService) DeleteExportRouteById(exportRouteId string) (*Response, error) {
```
Comment (loafoe) [inline cdl/export_route_service.go:175]: `DeleteExportRouteByID`

File: cdl/export_route_service_test.go [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+			},
+		},
+	}
+
+	muxCDL.HandleFunc("/store/cdl/"+cdlTenantID+"/ExportRoute", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		switch r.Method {
+		case "POST":
+			w.WriteHeader(http.StatusCreated)
+			io.WriteString(w, `{
```
Comment (loafoe) [inline cdl/export_route_service_test.go]: Add `_, _ =` to remove check failure

**Review by loafoe — CHANGES_REQUESTED** [review]

File: cdl/export_route_service.go [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+	Relation string `json:"relation"`
+	Url      string `json:"url"`
+}
+
+type ExportRouteBundleResponse struct {
+	ResourceType string                   `json:"resourceType,omitempty"`
+	Id           string                   `json:"id,omitempty"`
+	Type         string                   `json:"type,omitempty"`
+	Link         []LinkElementType        `json:"link,omitempty"`
+	Entry        []ExportRouteBundleEntry `json:"entry,required"`
```
Comment (loafoe) [inline cdl/export_route_service.go:93]: move `required` keyword to  `validate` label

File: cdl/export_route_service.go (hunk 1 above)
Comment (roopesh83) [reply cdl/export_route_service.go]: Changed CdlResearchStudy -> CDLResearchStudy

File: cdl/export_route_service.go (hunk 2 above)
Comment (roopesh83) [reply cdl/export_route_service.go]: Renamed CdlResearchStudy -> CDLResearchStudy

File: cdl/export_route_service.go (hunk 3 above)
Comment (roopesh83) [reply cdl/export_route_service.go:58]: Renamed CdlServiceAccount -> CDLServiceAccount

File: cdl/export_route_service.go (hunk 4 above)
Comment (roopesh83) [reply cdl/export_route_service.go]: Changed Url -> URL

File: cdl/export_route_service.go (hunk 5 above)
Comment (roopesh83) [reply cdl/export_route_service.go]: Renamed DeleteExportRouteById ->  DeleteExportRouteByID

File: cdl/export_route_service_test.go (hunk 6 above)
Comment (roopesh83) [reply cdl/export_route_service_test.go]: Added _, __ =

File: cdl/export_route_service.go (hunk 7 above)
Comment (roopesh83) [reply cdl/export_route_service.go]: Thanks! I have corrected this to include it within validate tag

File: cdl/export_route_service.go [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+	Name             string `json:"name,omitempty"`
+	ApprovalRequired bool   `json:"approvalRequired,omitempty"`
+}
+
+type ExportServiceAccount struct {
+	CDLServiceAccount ExportServiceAccountDetails `json:"cdlServiceAccount" validate:"required"`
+}
+
+type ExportServiceAccountDetails struct {
+	ServiceId           string `json:"serviceId" validate:"required"`
```
Comment (loafoe) [inline cdl/export_route_service.go:62]: sorry, missed this one: `ServiceID`

File: cdl/export_route_service.go [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+}
+
+type LinkElementType struct {
+	Relation string `json:"relation"`
+	URL      string `json:"url"`
+}
+
+type ExportRouteBundleResponse struct {
+	ResourceType string                   `json:"resourceType,omitempty"`
+	Id           string                   `json:"id,omitempty"`
```
Comment (loafoe) [inline cdl/export_route_service.go:90]: and this one, `ID`

**Review by loafoe — APPROVED** [review]

File: cdl/export_route_service.go (hunk 8 above)
Comment (roopesh83) [reply cdl/export_route_service.go]: ServiceId -> ServiceID

File: cdl/export_route_service.go (hunk 9 above)
Comment (roopesh83) [reply cdl/export_route_service.go]: Id -> ID

**Pull request merged**